File: MS.NET/clrzbus/examples/pool/PoolWorker.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using zbus;
namespace zbus
{
    class MyServiceHandler : ServiceHandler
    {
        public ZMsg HandleRequest(ZMsg request)
        {
            request.Dump();
            request.PushBack("from C# pool worker");

            return request;
        }
    }

    class PoolWorker
    {
        public static void Main(string[] args)
        {
            WorkerPoolConfig config = new WorkerPoolConfig();
            config.Service = "helloworld"; 

            WorkerPool pool = new WorkerPool(config);
            pool.Run(2, new MyServiceHandler()); //run with 2 threads
             
        } 
    }
}

File: trunk/JAVA/jzbus/src/test/java/net/zbus/pool/PoolWorker.java
package net.zbus.pool;

import net.zbus.ServiceHandler;
import net.zbus.Worker;
import net.zbus.WorkerHandler;
import net.zbus.WorkerPool;
import net.zbus.WorkerPoolConfig;
import net.zbus.ZMsg;

public class PoolWorker {
 
	public static void main(String[] args) { 
		WorkerPoolConfig cfg = new WorkerPoolConfig();
		cfg.setService("helloworld");
		cfg.setBrokers(new String[]{"127.0.0.1:15555","172.24.180.27:15555"});
		
		WorkerPool wc = new WorkerPool(cfg); 
		 
		wc.run(2, new ServiceHandler() { 			
			@Override
			public ZMsg handleRequest(ZMsg request) {
				request.dump(); 
				return request;
			} 
		});
	}

	public static void main2(String[] args) { 
		WorkerPoolConfig cfg = new WorkerPoolConfig();
		cfg.setService("helloworld"); 
		
		WorkerPool wc = new WorkerPool(cfg); 
		wc.run(2, new WorkerHandler() { 
			@Override
			public void handle(Worker worker) {
				while(true){
					ZMsg request = worker.recv();
					if(request == null) break;
					worker.send(request); 
				}
			}
		});
	}
}
